easybuild/tools/github.py

```diff
@@ -377,16 +378,34 @@ def download_repo(repo=GITHUB_EASYCONFIGS_REPO, branch='master', account=GITHUB_
     return extracted_path
 
 
-def fetch_easyblocks_from_pr(pr, path=None, github_user=None):
-    """Fetch patched easyconfig files for a particular PR."""
-    return fetch_files_from_pr(pr, path, github_user, github_repo=GITHUB_EASYBLOCKS_REPO)
+def pr_files_cache(func):
+    """
+    Decorator to cache result of fetch_files_from_pr.
+    """
+    cache = {}
 
+    @functools.wraps(func)
+    def cache_aware_func(pr, *args, **kwargs):
+        """Retrieve cached resul, or fetch files from PR & cache result."""
+        # cache key is combination of all function arguments (incl. optional ones)
+        key = tuple([pr] + [kwargs[key] for key in sorted(kwargs.keys())])
 
-def fetch_easyconfigs_from_pr(pr, path=None, github_user=None):
-    """Fetch patched easyconfig files for a particular PR."""
-    return fetch_files_from_pr(pr, path, github_user, github_repo=GITHUB_EASYCONFIGS_REPO)
+        if key in cache and all(os.path.exists(x) for x in cache[key]):
+            _log.debug("Using cached value for fetch_files_from_pr for PR #%s (%s)", pr, kwargs)
+            return cache[key]
+        else:
+            res = func(pr, *args, **kwargs)
+            cache[key] = res
+            return res
+
+    # expose clear/update methods of cache to wrapped function
+    cache_aware_func.clear_cache = cache.clear
+    cache_aware_func.update_cache = cache.update
```

Comment:
just to check if I understand correctly, these two methods are not (yet) used anywhere, right?

Reply:
Indeed, they're not, but they could come in useful at some point.

In the enhanced tests for example (which are still missing in this PR).

Reply:
fixed, tests are using `clear_cache` and `update_cache` now